muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/rest/WorkspaceRESTService.java

```diff
@@ -1707,7 +1721,11 @@ public void write(OutputStream out) throws IOException {
             
             ZipEntry ze = new ZipEntry(fileName);
             zout.putNextEntry(ze);
-            InputStream input = new ByteArrayInputStream(file.getContent());
+            byte[] content = file.getContent();
+            if (content == null) {
+              content = fileAnswerUtils.getFileContent(workspaceMaterialReply.getUserEntityId(), file.getFileId());
```

Comment:
getFileContent currently may return byte[0] if not changed - should this be checked?